worlds/smw/Aesthetics.py

```diff
@@ -160,21 +614,21 @@ def generate_shuffled_ow_music(world, player):
     shuffled_ow_music = ow_music_value_data.copy()
 
     if world.music_shuffle[player] == "consistent" or world.music_shuffle[player] == "full":
-        world.random.shuffle(shuffled_ow_music)
+        world.per_slot_randoms[player].shuffle(shuffled_ow_music)
     elif world.music_shuffle[player] == "singularity":
-        single_song = world.random.choice(shuffled_ow_music)
+        single_song = world.per_slot_randoms[player].choice(shuffled_ow_music)
         shuffled_ow_music = [single_song for i in range(len(shuffled_ow_music))]
 
     return shuffled_ow_music
 
 def generate_shuffled_ow_palettes(rom, world, player):
     if world.overworld_palette_shuffle[player]:
         for address, valid_palettes in valid_ow_palettes.items():
-            chosen_palette = world.random.choice(valid_palettes)
+            chosen_palette = world.per_slot_randoms[player].choice(valid_palettes)
             rom.write_byte(address, chosen_palette)
 
 def generate_shuffled_header_data(rom, world, player):
-    if world.music_shuffle[player] != "full" and not world.foreground_palette_shuffle[player] and not world.background_palette_shuffle[player]:
+    if world.music_shuffle[player] != "full" and world.foreground_palette_shuffle[player] == "on_legacy" and world.background_palette_shuffle[player] == "on_legacy" :
```

Comment:
We should early-out here when the shuffles are off, right? So maybe something like

```suggestion
    if world.music_shuffle[player] != "full" and world.foreground_palette_shuffle[player] == "off" and world.background_palette_shuffle[player] == "off" :
```

Reply:
Nvm, restructuring this following Discord conversation.